include/kodlab_mjbots_sdk/mjbots_robot_interface.h

```diff
@@ -124,6 +134,7 @@ class MjbotsRobotInterface {
   std::future<::mjbots::moteus::Pi3HatMoteusInterface::Output> can_result_;      /// future can result, used to check if
   /// response is ready
   SoftStart soft_start_;                                                      /// Soft Start object
+  ::mjbots::pi3hat::Attitude attitude_;                                       /// Robot attitude
```

Comment:
I think if you are going to use smart pointers, they should be smart pointers everywhere including in the owner. Otherwise we are defeating the point of them.

Reply:
So its important that this is a copy of the pi3hat::Attitude object that is passed into the pi3hat moteus interface other wise you get a race condition. For this object I don't ever use it as a smart ptr.